pkg/asset/internal/templates.go

```diff
@@ -513,4 +523,92 @@ spec:
     port: 2379
     protocol: TCP
 `)
+
+	KubeFlannelCfgTemplate = []byte(`apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kube-flannel-cfg
+  namespace: kube-system
+  labels:
+    tier: node
+    app: flannel
+data:
+  cni-conf.json: |
+    {
+      "name": "cbr0",
+      "type": "flannel",
+      "delegate": {
+        "isDefaultGateway": true
+      }
+    }
+  net-conf.json: |
+    {
+      "Network": "10.2.0.0/16",
+      "Backend": {
+        "Type": "vxlan"
+      }
+    }
+`)
+
+	KubeFlannelTemplate = []byte(`apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: kube-flannel
+  namespace: kube-system
+  labels:
+    tier: node
+    app: flannel
+spec:
+  template:
+    metadata:
+      labels:
+        tier: node
+        app: flannel
+    spec:
+      hostNetwork: true
+      containers:
+      - name: kube-flannel
+        image: quay.io/coreos/flannel-git:v0.6.1-60-g4b752d1-amd64
+        command: [ "/opt/bin/flanneld", "--ip-masq", "--kube-subnet-mgr", "--iface=$(POD_IP)"]
+        securityContext:
+          privileged: true
+        env:
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: POD_IP
```

Comment:
We normally would have set this via an environment file on the host (e.g. https://github.com/kubernetes-incubator/bootkube/blob/v0.3.0/hack/quickstart/init-master.sh#L38)

Flannel will guess which interface based on the default route -- unfortunately, the default route for virtualbox is an interface which is not routable between nodes.

Not sure on AWS/GCE -- but I'm also assuming you would want to use the private network (when using hostNetwork + podIP -- this will be the IP of the interface which is used as the node name).